app/modules/auth/sso_providers/azure_provider.py

```
@@ -0,0 +1,185 @@
+                provider_uid=payload["oid"],  # Object ID
+                display_name=payload.get("name"),
+                given_name=payload.get("given_name"),
+                family_name=payload.get("family_name"),
+                raw_data=payload,
+            )
+
+            logger.info(f"Successfully verified Azure AD token for {user_info.email}")
+            return user_info
+
+        except jwt.ExpiredSignatureError:
+            logger.error("Azure AD token has expired")
```

Ruff (0.14.7)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)